extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmCdiProcessor.java

```diff
@@ -245,6 +253,39 @@ void generateDataSourceBeans(HibernateOrmRecorder recorder,
         }
     }
 
+    @BuildStep
+    void registerBeans(HibernateOrmConfig hibernateOrmConfig,
+            BuildProducer<AdditionalBeanBuildItem> additionalBeans,
+            BuildProducer<UnremovableBeanBuildItem> unremovableBeans,
+            Capabilities capabilities,
+            CombinedIndexBuildItem combinedIndex,
+            List<PersistenceUnitDescriptorBuildItem> descriptors,
+            JpaModelBuildItem jpaModel) {
+        if (!HibernateOrmProcessor.hasEntities(jpaModel)) {
+            return;
+        }
+
+        if (capabilities.isMissing(Capability.HIBERNATE_REACTIVE)) {
+            // The following beans only make sense for Hibernate ORM, not for Hibernate Reactive
+            List<Class<?>> unremovableClasses = new ArrayList<>();
+            if (capabilities.isPresent(Capability.TRANSACTIONS)) {
+                unremovableClasses.add(TransactionManager.class);
+                unremovableClasses.add(TransactionSessions.class);
+            }
+            unremovableClasses.add(RequestScopedSessionHolder.class);
+            unremovableClasses.add(RequestScopedStatelessSessionHolder.class);
+            unremovableClasses.add(QuarkusArcBeanContainer.class);
```

Comment:
This class makes sense even for Hibernate Reactive.

Reply:
Ok, let's fix this one. Out of curiosity, what is this bean used for in HR?

Reply:
Same as in Hibernate ORM: to fetch pluggable components (e.g. `AttributeConverter`) implemented as CDI beans.